components/layout_2020/flow/inline.rs

```diff
@@ -2407,3 +2410,27 @@ fn char_prevents_soft_wrap_opportunity_when_before_or_after_atomic(character: ch
         class == XI_LINE_BREAKING_CLASS_WJ ||
         class == XI_LINE_BREAKING_CLASS_ZWJ
 }
+
+fn to_length(value: &LogicalVec2<CSSPixelLength>) -> LogicalVec2<Au> {
+    LogicalVec2 {
+        inline: value.inline.into(),
+        block: value.block.into(),
+    }
+}
+
+fn to_au_rect(value: LogicalRect<Au>) -> LogicalRect<CSSPixelLength> {
+    LogicalRect {
+        start_corner: LogicalVec2 {
+            inline: value.start_corner.inline.into(),
+            block: value.start_corner.block.into(),
+        },
+        // If this float has a negative margin, we should only consider its non-negative
+        // block size contribution when determing where to place it. When the margin is
+        // so negative that it's placed completely above the current float ceiling, then
+        // we should position it as if it had zero block size.
```

Comment:
I'm not sure this comment belongs here? It doesn't really have anything to do with conversion of units.

Reply:
right, i have removed it.